File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} root1
# @param {TreeNode} root2
# @return {Integer[]}
def print_elem(root, a)
    return a if root == nil
    print_elem(root.left, a)
    a.append(root.val)
    print_elem(root.right, a)
end
def get_all_elements(root1, root2)
    a=[]
    b=print_elem(root1,a)
    puts "hello"
    b=print_elem(root2,a)
    b.sort
end

File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    vector<int> sol;
public:
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        printElem(root1);
        printElem(root2);
        sort(sol.begin(),sol.end());
        return sol;
    }
    void printElem(TreeNode* root){
        if (!root) return;
        printElem(root->left);
        sol.push_back(root->val);
        printElem(root->right);
        
    }
};